src/main/java/book/store/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,42 @@
 package book.store.service.impl;
 
+import book.store.dto.BookDto;
+import book.store.dto.CreateBookRequestDto;
+import book.store.exception.EntityNotFoundException;
+import book.store.mapper.BookMapper;
 import book.store.model.Book;
 import book.store.repository.BookRepository;
 import book.store.service.BookService;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        Book savedBook = bookRepository.save(book);
+        return bookMapper.toDto(savedBook);
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public Optional<BookDto> findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book with id: " + id)
+        );
+        return Optional.ofNullable(bookMapper.toDto(book));
```

Comment:
Inline expression using Optional#map
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(
                () -> new EntityNotFoundException("Can't find book with id: " + id)
        );
        return Optional.ofNullable(bookMapper.toDto(book));
```